assets/app/views/_pods.html

```diff
@@ -2,21 +2,102 @@
 Expects the following variables:
 pods - a hash of pod objects by id
 -->
-<div class="component-block pod-container">
-  <div class="animate-repeat pod osc-object" ng-repeat="pod in pods track by (pod | uid)" kind="Pod" resource="pod"
-      ng-class="(pod | isTroubledPod) ? 'pod-warning' : ('pod-' + pod.status.phase.toLowerCase())">
-    <div class="pod-text" ng-switch="pod.status.phase">
-      <strong class="pod-status-label">{{pod.status.phase}}</strong>
-      <div ng-switch-when="Pending">
-        <span ng-if="!pod.spec.nodeName">scheduling...</span>
-        <span ng-if="pod.spec.nodeName && !pod.status.startTime">scheduled</span>
-        <span ng-if="pod.spec.nodeName && pod.status.startTime">pulling...</span>
+<div class="component-block" ng-if="(pods | hashSize) > 3">
+  <div class="pod-container">
+    <div class="pod  pod-{{phase.toLowerCase()}} pod-multiple osc-object osc-object-stacked" ng-repeat-start="phase in phases"  ng-if="(phasePods = (pods | podsForPhase : phase)).length && (troublePods = (phasePods | troubledPods)) && (expandedPhase != phase || warningsExpanded) && (phasePods.length - troublePods.length > 0)" ng-click="expandPhase(phase, false, $event)" title="Expand to see individual pods">
+      <div style="font-size:24px; line-height: 19px;">
+        <span>{{phasePods.length - troublePods.length}}</span>
       </div>
-      <div ng-switch-default>
-        &nbsp;
-        <span ng-if="pod.status.podIP">{{pod.status.podIP}}</span>
-        <relative-timestamp ng-if="!pod.status.podIP" timestamp="pod.status.startTime"></relative-timestamp>
-        &nbsp;
+      <div class="pod-text">{{phase}}</div>
+      <i class="fa fa-ellipsis-h"></i>
+    </div>
+    <div class="pod pod-warning pod-multiple osc-object osc-object-stacked" ng-repeat-end  ng-if="(expandedPhase != phase || !warningsExpanded) && (phasePods = (pods | podsForPhase : phase)).length && (troublePods = (phasePods | troubledPods)).length" ng-click="expandPhase(phase, true, $event)" title="Expand to see individual pods">
```

Comment:
Believe it or not, yes.  ng-init doesn't re-evaluate the same way, so it
wasn't getting updated when the pods map was updating

On Thu, Sep 3, 2015 at 11:06 AM, Sam Padgett <notifications@github.com>
wrote:

> In assets/app/views/_pods.html
> <https://github.com/openshift/origin/pull/4486#discussion_r38656200>:
>
> >        </div>
> > -      <div ng-switch-default>
> > -        &nbsp;
> > -        <span ng-if="pod.status.podIP">{{pod.status.podIP}}</span>
> > -        <relative-timestamp ng-if="!pod.status.podIP" timestamp="pod.status.startTime"></relative-timestamp>
> > -        &nbsp;
> > +      <div class="pod-text">{{phase}}</div>
> > +      <i class="fa fa-ellipsis-h"></i>
> > +    </div>
> > +    <div class="pod pod-warning pod-multiple osc-object osc-object-stacked" ng-repeat-end  ng-if="(expandedPhase != phase || !warningsExpanded) && (phasePods = (pods | podsForPhase : phase)).length && (troublePods = (phasePods | troubledPods)).length" ng-click="expandPhase(phase, true, $event)" title="Expand to see individual pods">
>
> Not crazy about assignments inside an ng-if. Any reason not use to ng-init
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4486/files#r38656200>.
>